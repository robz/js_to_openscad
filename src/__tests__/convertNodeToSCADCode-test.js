/* @flow */

declare var describe: any;
declare var it: any;
declare var expect: any;

describe('convertNodeToSCADCode', () => {
  it('does stuff', () => {
    const OpenSCAD = require('../OpenSCAD');
    const convertNodeToSCADCode = require('../convertNodeToSCADCode');

    const s = new OpenSCAD();

    function symY(dy, child) {
      return s.union(...[-1, 1].map(
        dir => s.translate([0, dir * dy, 0]).thing(child)
      ));
    };

    const head = s
      .rotate([0, 15, 0])
      .difference(
        s.hull(
          s.sphere({r: .6}),
          s.translate([.5, 0, 0]).sphere({r: .3})
        ),
        symY(
          .5,
          s.translate([.2, 0, .2]).sphere({r: .11}),
        )
      );

    expect(convertNodeToSCADCode(head)).toMatchSnapshot();
  });
});
