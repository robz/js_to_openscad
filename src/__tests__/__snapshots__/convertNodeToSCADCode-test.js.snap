exports[`convertNodeToSCADCode does stuff 1`] = `
"rotate([0, 15, 0])
difference() {
  hull() {
    sphere(r = 0.6);

    translate([0.5, 0, 0])
    sphere(r = 0.3);
  }

  union() {
    translate([0, -0.5, 0])
    translate([0.2, 0, 0.2])
    sphere(r = 0.11);

    translate([0, 0.5, 0])
    translate([0.2, 0, 0.2])
    sphere(r = 0.11);
  }
}"
`;
